index.js

```diff
@@ -21,72 +37,98 @@ var resultStrings = [
   'MISMATCH'
 ];
 
-exports.resultCodeToString = function(code) {
-  if (code < 1 || code >= resultStrings.length) { return; }
-  return resultStrings[code];
+HmacAuth.resultCodeToString = function(code) {
+  return resultStrings[code - 1];
 };
 
 function signedHeaders(req, headers) {
-  return headers.map(function(header) { return req.get(header) || ''; });
+  return headers.map(function(header) {
+    var value = req.headers[header];
+    if (typeof value === Array) { value = value.join(','); }
```

Comment:
If `value` is an array, `typeof value` would return `'object'` (a string). How about testing for what you intend to use, in this case:
```js
if (typeof value !== 'sting' && typeof value.join === 'function') { 
  value = value.join(','); 
}
```